solc/CommandLineInterface.cpp

```diff
@@ -1013,34 +1013,54 @@ bool CommandLineInterface::assemble(yul::AssemblyStack::Language _language, yul:
 
 		yul::AssemblyStack& stack = assemblyStacks[src.first];
 
-		sout() << endl << "Pretty printed source:" << endl;
-		sout() << stack.print() << endl;
+		if (m_options.compiler.outputs.irOptimized)
+		{
+			// NOTE: This actually outputs unoptimized code when the optimizer is disabled but
+			// 'ir' output in StandardCompiler works the same way.
+			sout() << endl << "Pretty printed source:" << endl;
+			sout() << stack.print() << endl;
+		}
 
 		if (_language != yul::AssemblyStack::Language::Ewasm && _targetMachine == yul::AssemblyStack::Machine::Ewasm)
 		{
 			stack.translate(yul::AssemblyStack::Language::Ewasm);
 			stack.optimize();
 
-			sout() << endl << "==========================" << endl;
-			sout() << endl << "Translated source:" << endl;
-			sout() << stack.print() << endl;
+			// TODO: This isn't ewasm but it's only present when we're doing Yul->EWASM translation.
```

Comment:
I guess this is not going to be done in this PR? Maybe create an issue so it doesn't get forgotten?

Reply:
Will do.

Reply:
Actually, I just realized that ewasm is still experimental. So maybe it's not breaking after all?

Reply:
Actually, I just realized that ewasm is still experimental. So maybe it's not breaking after all?

Reply:
Got a confirmation on the chat. It's not breaking.

In that case I'm adding an extra commit that adds an output called `--ewasm-ir` (I could not come up with a better name).